src/rosegold/control/inventory.cr

```diff
@@ -75,19 +75,71 @@ class Rosegold::Inventory
   #   inventory.deposit_at_least 5, "diamond_pickaxe" # => 3
   #   inventory.deposit_at_least 5, &.empty?, hotbar # => 1
   #   inventory.deposit_at_least 5, { |slot| slot.item_id == "diamond_pickaxe" && slot.efficiency >= 4 } # => 2
-  def deposit_at_least(count, spec, source : Array(WindowSlot) = inventory + hotbar)
-    shift_click_at_least count, spec, source
+  def deposit_at_least(count, spec, source : Array(WindowSlot) = inventory + hotbar, target : Array(WindowSlot) = content)
+    transferred = 0
+  
+    # prefer large stacks for minimum clicks
+    # for equal stacks, preserve order
+    source.sort_by { |s| s.count }.each do |slot|
```

Comment:
If you decide to start with small stacks (default sort order is ascending), the comment should be changed/removed too.